### PR TITLE
[2.7][Asset] Ability to set empty version strategy in packages

### PR DESCRIPTION
**Pull request by ewgRa**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no, but not sure, as for me test was wrong |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14832 |
| License | MIT |
| Doc PR |  |

Comment about '' thing. As you can see in xml test - we can't for attribute set null value. And for xml version '' empty string considered as null value, that affect all others formats and test is changed.

### CODE REVIEW

**Comment by xabbuh** [comment]
Won't the version be `null` when you omitted the `version` attribute in the XML format (instead of setting it to an empty string)?

**Comment by ewgRa** [comment]
There is a case, if version is omitted - then default strategy used.

If we change it and if in xml omitted version considered as empty strategy (also as we need sync same logic for php, yml then) - we will have broken BC.

There is also legacy code (framework templating configuration) that have different logic - omitted considered as empty.

**Comment by ewgRa** [comment]
There is no possible to set attribute in xml to null as far as I know :(

**Comment by XWB** [comment]
Ping @fabpot Any feedback on this? This bug was introduced with Symfony 2.7 and has not been resolved yet.

**Comment by ewgRa** [comment]
ping @fabpot @xabbuh any comments?

**Comment by fabpot** [comment]
:+1:

@ewgRa Can you rebase to get rid of the 2 merge commit that prevents me from merging? Or even better, squash all commits. Thanks.

**Comment by ewgRa** [comment]
@fabpot done

**Comment by fabpot** [comment]
Thank you @ewgRa.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -547,7 +553,12 @@ private function addAssetsSection(ArrayNodeDefinition $rootNode)
                             ->prototype('array')
                                 ->fixXmlConfig('base_url')
                                 ->children()
-                                    ->scalarNode('version')->defaultNull()->end()
```
Comment (ewgRa) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: @fabpot Sorry, I accidently omit defaultNull here - fix #17514

**Pull request merged**